hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/BalancerConditionals.java

```diff
@@ -90,8 +94,20 @@ boolean isReplicaDistributionEnabled() {
       .anyMatch(DistributeReplicasConditional.class::isAssignableFrom);
   }
 
-  boolean shouldSkipSloppyServerEvaluation() {
-    return isConditionalBalancingEnabled();
+  boolean isTableIsolationEnabled() {
+    return conditionalClasses.contains(MetaTableIsolationConditional.class);
+  }
+
+  boolean isServerHostingIsolatedTables(BalancerClusterState cluster, int serverIdx) {
+    Set<TableIsolationConditional> tableIsolationConditionals =
+      conditionals.stream().filter(TableIsolationConditional.class::isInstance)
+        .map(TableIsolationConditional.class::cast).collect(Collectors.toSet());
+    for (TableIsolationConditional tableIsolationConditional : tableIsolationConditionals) {
```

Comment:
nit: you're already Streaming, so just keep streaming and return the result of `anyMatch` or `findAny`.

I often find the transition between Stream and iteration jarring...